src/dialog/dialog.service.ts

```diff
@@ -80,59 +107,106 @@ export class DialogService {
+		let focusedElement = document.activeElement;
+		dialogConfig["previouslyFocusedElement"] = focusedElement;
+		dialogRef.instance.dialogConfig = dialogConfig;
+		this.onClose = dialogRef.instance.close;
+		this.isOpen = true;
+
+		const closeSubscription = this.onClose.subscribe(() => {
+			if (dialogConfig.shouldClose && dialogConfig.shouldClose()) {
+				this.close(viewContainer, dialogRef);
 			}
+		});
 
-			// initialize some extra options
-			let focusedElement = document.activeElement;
-			dialogConfig["previouslyFocusedElement"] = focusedElement;
-			this.dialogRef.instance.dialogConfig = dialogConfig;
-			this.onClose = this.dialogRef.instance.close;
-			this.isOpen = true;
+		// keep track of all dialog subscriptions
+		DialogService.dialogCloseSubscription.add(closeSubscription);
 
-			this.dialogSubscription = this.onClose.subscribe(() => {
-				this.close(viewContainer);
-			});
+		dialogRef.instance.elementRef.nativeElement.focus();
 
-			this.dialogRef.instance.elementRef.nativeElement.focus();
-		}
+		// deprecated - kept for API compat
+		this.dialogRef = dialogRef;
+
+		// deprecated - kept for API compat
+		this.dialogSubscription = closeSubscription;
+
+		// return `this` for easy method chaining
+		// TODO v11: return `dialogRef` instead
 		return this;
 	}
 
 	/**
 	 * On close of `Dialog` item, sets focus back to previous item, unsets
 	 * the current `dialogRef` item. Unsubscribes to the event of `Dialog` close.
+	 *
+	 * @param viewContainer deprecated - does nothing. null may safely be passed
+	 * @param dialogRef the dialogRef to close
 	 */
-	close(viewContainer: ViewContainerRef) {
+	close(viewContainer: ViewContainerRef, dialogRef: ComponentRef<Dialog> = this.dialogRef) {
 		this.isClosed.emit(true);
 
-		if (this.dialogRef) {
-			let elementToFocus = this.dialogRef.instance.dialogConfig["previouslyFocusedElement"];
-			if (this.placeholderService.hasPlaceholderRef() && !this.dialogRef.instance.dialogConfig.appendInline) {
-				this.placeholderService.destroyComponent(this.dialogRef);
-			} else {
-				viewContainer.remove(viewContainer.indexOf(this.dialogRef.hostView));
-			}
-			this.dialogRef = null;
-			this.isOpen = false;
-			elementToFocus.focus();
+		// to handle the case where we have a null `this.dialogRef`
+		if (!dialogRef) { return; }
 
-			if (this.dialogSubscription) {
-				this.dialogSubscription.unsubscribe();
-			}
+		let elementToFocus = dialogRef.instance.dialogConfig["previouslyFocusedElement"];
+
+		dialogRef.destroy();
+
+		if (DialogService.dialogRefs.has(dialogRef)) {
+			DialogService.dialogRefs.delete(dialogRef);
 		}
+
+		this.dialogRef = null;
+		this.isOpen = false;
+		elementToFocus.focus();
+
+		if (this.dialogSubscription) {
+			this.dialogSubscription.unsubscribe();
+		}
+	}
+
+	/**
+	 * Closes all known `Dialog`s. Does not focus any previous elements, since we can't know which would be correct
+	 */
+	closeAll() {
+		DialogService.dialogRefs.forEach(ref => {
```

Comment:
Having it in one line, no curlies, could improve readability.

Reply:
Yeah, in this case it's short enough there's no need to split it up